saleor/graphql/product/filters.py

```diff
@@ -1246,20 +1229,16 @@ def filter_available_from(self, queryset, name, value):
 
     def filter_variant_price(self, qs, _, value):
         channel_slug = get_channel_slug_from_filter_data(self.data)
-        channel_id = Channel.objects.using(qs.db).filter(slug=channel_slug).values("pk")
-        variant_listing = ProductVariantChannelListing.objects.using(qs.db).filter(
-            Exists(channel_id.filter(pk=OuterRef("channel_id")))
-        )
-        variant_listing = filter_where_by_numeric_field(
-            variant_listing, "price_amount", value
-        )
-        variant_listing = variant_listing.values("variant_id")
-        variants = (
-            ProductVariant.objects.using(qs.db)
-            .filter(Exists(variant_listing.filter(variant_id=OuterRef("pk"))))
-            .values("product_id")
-        )
-        return qs.filter(Exists(variants.filter(product_id=OuterRef("pk"))))
+        channel = Channel.objects.using(qs.db).filter(slug=channel_slug).first()
+        if not channel:
```

Comment:
Are you sure that your solution will behave in the same way for not existing slug?